mapper/aggregateMapper.go

```diff
@@ -31,25 +32,37 @@ func (m *DefaultCPHAggregateMapper) MapContentPlaceholder(mpc *model.MethodeCont
 	if err != nil {
 		return nil, err
 	}
-
 	uuid := ""
-	if m.isBlogCategory(mpc) {
-		resolvedUuid, err := m.iResolver.ResolveIdentifier(mpc.Attributes.ServiceId, mpc.Attributes.RefField, tid)
+
+	if m.isGenericContent(mpc) {
+		resolvedUUID, err := gouuid.FromString(mpc.Attributes.OriginalUUID)
+		if err != nil {
+			return nil, fmt.Errorf("invalid generic uuid: %v", err)
+		}
+		uuid = resolvedUUID.String()
+		found, err := m.iResolver.ContentExists(uuid, tid)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't check OriginalUUID in document store: %v", err)
+		}
+		if !found {
+			return nil, fmt.Errorf("couldn't find OriginalUUID %s in document store", uuid)
+		}
+	} else if m.isBlogCategory(mpc) {
+		uuid, err = m.iResolver.ResolveIdentifier(mpc.Attributes.ServiceId, mpc.Attributes.RefField, tid)
```

Comment:
We should probably check if the fields `ServiceId` and `RefField` are populated before sending them further.